ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -160,74 +217,120 @@ impl Blockstore {
             .db
             .batch()
             .expect("Database Error: Failed to get write batch");
+
         let mut delete_range_timer = Measure::start("delete_range");
+        let columns_purged = self.purge_range(&mut write_batch, from_slot, to_slot, purge_type)?;
+        delete_range_timer.stop();
+
+        let mut write_timer = Measure::start("write_batch");
+        if let Err(e) = self.db.write(write_batch) {
+            error!(
+                "Error: {:?} while submitting write batch for slot {:?} retrying...",
+                e, from_slot
+            );
+            return Err(e);
+        }
```

Comment:
Similar to other comment:
- I think you might be able to simplify slightly with `.inspect_err()` + `?`
- That `retrying` bit isn't accurate, so can you adjust the message. Can you also adjust the message to include `from_slot` and `to_slot`